diff --git a/tests/Config/TestContainersWebTestCase.php b/tests/Config/TestContainersWebTestCase.php
new file mode 100644
index 0000000..0389d98
--- /dev/null
+++ b/tests/Config/TestContainersWebTestCase.php
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Config;
+
+use Testcontainers\Modules\PostgresContainer;
+use Testcontainers\Container\GenericContainer;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Testcontainers\Container\StartedGenericContainer;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+
+class TestContainersWebTestCase extends WebTestCase
+{
+    protected static ?StartedGenericContainer $postgresContainer = null;
+    protected static ?StartedGenericContainer $natsContainer     = null;
+
+    public static function createClient(array $options = [], array $server = []): KernelBrowser
+    {
+        $kernel = self::$kernel;
+        $client = $kernel->getContainer()->get('test.client');
+        $client->setServerParameters($server);
+
+        return self::getClient($client);
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->postgresContainer = new PostgresContainer(
+            version: 'latest',
+            username: 'app_user',
+            password: 's3cr3t',
+            database: 'app_test',
+        )
+            ->withExposedPorts(5432)
+            ->start();
+
+        $postgresHost = $this->postgresContainer->getHost();
+        $postgresPort = $this->postgresContainer->getFirstMappedPort();
+
+        $databaseUrl = "postgres://app_user:s3cr3t@{$postgresHost}:{$postgresPort}/app_test?serverVersion=18&charset=utf8";
+
+        $_ENV['DATABASE_URL'] = $databaseUrl;
+
+        $this->natsContainer = new GenericContainer(image: 'nats:latest')
+            ->withEntryPoint('/nats-server')
+            ->withCommand(['--config', '/nats-server.conf', '--user', 'app_user', '--pass', 's3cr3t'])
+            ->withExposedPorts(4222)
+            ->start();
+
+        $natsHost = $this->natsContainer->getHost();
+        $natsPort = $this->natsContainer->getFirstMappedPort();
+
+        $_ENV['NATS_HOST'] = $natsHost;
+        $_ENV['NATS_PORT'] = $natsPort;
+        $_ENV['NATS_USER'] = 'app_user';
+        $_ENV['NATS_PASS'] = 's3cr3t';
+
+        self::bootKernel();
+
+        $application = new Application(self::$kernel);
+        $application->setAutoExit(false);
+
+        $runMigrationsInput = new ArrayInput([
+            'command'          => 'doctrine:migrations:migrate',
+            '--no-interaction' => true,
+            '--all-or-nothing' => true,
+        ]);
+
+        $application->run($runMigrationsInput, new NullOutput());
+
+        $runFixturesInput = new ArrayInput([
+            'command'               => 'doctrine:fixtures:load',
+            '--no-interaction'      => true,
+            '--purge-with-truncate' => true,
+        ]);
+
+        $application->run($runFixturesInput, new NullOutput());
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->postgresContainer?->stop();
+        $this->natsContainer?->stop();
+    }
+}
